### ISSUE TITLE
Is it possible to use feature map loss?

### ISSUE THREAD

**Issue by John1231983** [junyanz/pytorch-CycleGAN-and-pix2pix#792]
The feature map loss shows some advantages for new version of pix2pix. Do you think it also useful when we use the loss for cyclegan? Because it is the unpaired image, which images should we use for input of feature map: idt syn or reconstruction images? Thanks

**Comment by junyanz** [comment]
Do you mean the feature matching loss (or perceptual loss)?  The matching loss requires paired data. You may want to check out a recent [work](http://openaccess.thecvf.com/content_ECCV_2018/papers/Diana_Sungatullina_Image_Manipulation_with_ECCV_2018_paper.pdf) that tries to use a hybrid of feature matching and GAN loss in unpaired cases.

**Comment by John1231983** [comment]
Yes. The feature maching loss that is I meant. Thanks for your reference.

**Comment by weightofname** [comment]
@junyanz Im,sorry to disturb you.If we input the fake_B and the real_B as the paried data,can we use the Perceptual loss in Cyclegan?

**Comment by junyanz** [comment]
Yeah, you can use it. You may also consider using the pix2pix data loader since you have paired dataset.

**Comment by weightofname** [comment]
thanks for you reply! i have another question is if i want to increase low dimensional features of the target image domain in the converted image,where can i use the Perceptual loss in the cyclegan model ? i once read a document said add the  Perceptual loss in the Discriminator net to improve network performance, but it no open source and  Explain the structure carefully. hope you can answer my questions!

**Comment by junyanz** [comment]
Using perceptual loss and GANs loss together: please refer to [pix2pixHD](https://github.com/NVIDIA/pix2pixHD) (also used in [SPADE/GauGAN](https://github.com/NVlabs/SPADE)). Perceptual loss can stabilize GANs training. 

Use GAN loss on top of pre-trained features: please refer to percpetual discriminator [paper](https://arxiv.org/abs/1809.01396). It is used in recent GANs papers ([projected-gans](https://arxiv.org/abs/2111.01007) and [vision-aided gans](https://www.cs.cmu.edu/~vision-aided-gan/)) as well.
